src/shared/sam/debugger/typescriptSamDebug.ts

```diff
@@ -113,44 +113,72 @@ export async function makeTypescriptConfig(config: SamLaunchRequestArgs): Promis
-        const tscResponse = await new ChildProcess(true, 'tsc', undefined, '-v').run()
-        if (tscResponse.exitCode !== 0 || !tscResponse.stdout.startsWith('Version')) {
-            throw new Error('TypeScript compiler "tsc" not found.')
-        }
-        const samBuildOutputAppRoot = path.join(
-            config.baseBuildDir!,
-            'output',
-            path.parse(config.invokeTarget.projectRoot).name
-        )
-        const tsconfigPath = path.join(samBuildOutputAppRoot, 'tsconfig.json')
-        if (
-            (await readdir(config.codeRoot)).includes('tsconfig.json') ||
-            (await hasFileWithSuffix(config.codeRoot, '.ts', '**/node_modules/**'))
-        ) {
-            //  This default config is a modified version from the AWS Toolkit for JetBrain's tsconfig file. https://github.com/aws/aws-toolkit-jetbrains/blob/911c54252d6a4271ee6cacf0ea1023506c4b504a/jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/lambda/nodejs/NodeJsLambdaBuilder.kt#L60
-            const defaultTsconfig = {
-                compilerOptions: {
-                    target: 'es6',
-                    module: 'commonjs',
-                    typeRoots: ['node_modules/@types'],
-                    types: ['node'],
-                    rootDir: '.',
-                    inlineSourceMap: true,
-                },
-            }
-            try {
-                writeFileSync(tsconfigPath, JSON.stringify(defaultTsconfig, undefined, 4))
-                getLogger('channel').info('Compiling TypeScript')
-                await new ChildProcess(true, 'tsc', undefined, '--project', samBuildOutputAppRoot).run()
-            } catch (error) {
-                getLogger('channel').error(`Compile Error: ${error}`)
-                throw Error('Failed to compile typescript Lambda')
-            }
+    if (!config.baseBuildDir) {
+        throw Error('invalid state: config.baseBuildDir was not set')
+    }
+    if (config.invokeTarget.target !== 'code') {
+        return
+    }
+
+    async function findTsOrTsConfig(dir: string, child: boolean): Promise<string | undefined> {
+        const glob = (child ? '*/' : '') + '{*.ts,tsconfig.json}'
+        const found = await vscode.workspace.findFiles(new vscode.RelativePattern(dir, glob), '**/node_modules/**', 1)
+        if (found.length === 0) {
+            return undefined
         }
+        return found[0].fsPath
+    }
+
+    // Require tsconfig.json or *.ts in the top-level of the source app, to
+    // indicate a typescript Lambda. #2086
+    // Note: we don't use this tsconfig.json for compiling the target=code
+    // Lambda app below, instead we generate a minimal one.
+    const isTsApp = (await findTsOrTsConfig(config.codeRoot, false)) !== undefined
+    if (!isTsApp) {
+        return
+    }
+
+    const buildOutputDir = path.join(config.baseBuildDir, 'output')
+    const buildDirTsFile = await findTsOrTsConfig(buildOutputDir, true)
```

Comment:
Out of curiosity, is this check sufficient to cover both the `isTsApp` and `buildDirTsFile` cases?

Reply:
Not sure if I understand the question, but: isTsApp checks the source app dir, which is different than this buildDir case. This latter case is slightly less deterministic.

Reply:
If `buildOutputDir` doesn't exist, can we just assume this isn't a TS app and return? Do we need the two separate checks with this one throwing an error?

Reply:
I don't know, but the current granularity gives more information at a negligible cost, and avoids needing to answer that question.